packages/@n8n/db/src/migrations/sqlite/1758794506893-AddProjectIdToVariableTable.ts

```diff
@@ -0,0 +1,96 @@
+import type { MigrationContext, ReversibleMigration } from '../migration-types';
+const VARIABLES_TABLE_NAME = 'variables';
+const VARIABLES_TEMP_TABLE_NAME = 'variables_temp';
+const UNIQUE_PROJECT_KEY_INDEX_NAME = 'variables_project_key_unique';
+const UNIQUE_GLOBAL_KEY_INDEX_NAME = 'variables_global_key_unique';
+/**
+ * Adds a projectId column to the variables table to support project-scoped variables.
+ * Also updates the unique constraints to allow the same key to be used in different projects,
+ * while still enforcing uniqueness for global variables (where projectId is null).
+ */
+export class AddProjectIdToVariableTable1758794506893 implements ReversibleMigration {
+	async up({
+		schemaBuilder: { createTable, column, createIndex, dropTable },
+		queryRunner,
+		escape,
+		copyTable,
+	}: MigrationContext) {
+		const variablesTableName = escape.tableName(VARIABLES_TABLE_NAME);
+		const tempVariablesTableName = escape.tableName(VARIABLES_TEMP_TABLE_NAME);
+
+		await createTable(VARIABLES_TEMP_TABLE_NAME)
+			.withColumns(
+				column('id').varchar(36).primary.notNull,
+				column('key').text.notNull,
+				column('type').text.notNull.default("'string'"),
+				column('value').text,
+				column('projectId').varchar(36),
+			)
+			.withForeignKey('projectId', { tableName: 'project', columnName: 'id', onDelete: 'CASCADE' });
+
+		// Create unique index for project variables (projectId not null)
+		await createIndex(
```

Comment:
I would also imagine, that his covers the projectId null values right?

Reply:
Same explanation, same behavior from sqlite

Reply:
Ignore this, we do need the second index, my mistake.